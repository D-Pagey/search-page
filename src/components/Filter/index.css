.filter-container {
  width: 100%;
}

.back-to-search-div {
  align-items: center;
  background-color: #172B24;
  display: flex;
  height: 52px;
}

.chevron {
  margin-left: 20px;
}

.back-to-search {
  color: white;
  margin-left: 10px;
}

.filter-form {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 0 auto 40px;
  padding: 0 1rem;
}

.input-group {
  display: flex;
  flex-direction: column;
  margin-top: 15px;
  width: 100%;
}

.filter-label {
  color: #292b2c;
  font-size: 16px;
  font-weight: bold;
  margin: 0 0 .2rem 2px;
}

.filter-input {
  border: 1px solid #e4e3e3;
  border-radius: 6px;
  color: black;
  font-size: 100%;
  height: 40px;
  padding: 0 10px;
  width: 100%;
}