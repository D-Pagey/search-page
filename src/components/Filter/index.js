import React from 'react';
import {
  func,
  objectOf,
  any,
  string,
} from 'prop-types';

import './index.css';

// To Do: take user's input above into state to append to fetch url

export default function Filter({
  toggle,
  handleChange,
  defaultValue,
  className,
}) {
  return (
    <div className={className}>
      <div className="back-to-search-div">
        <svg className="chevron" onClick={toggle} id="Layer_4" data-name="Layer 4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 23 43.41" width="20px" height="20px"><title>white_chevron</title><rect x="-29.83" y="-19.96" width="83.33" height="83.33" rx="41.67" ry="41.67" fill="none" /><polyline points="21.17 1.71 11.83 11.71 2.5 21.71 11.83 31.71 21.17 41.71" fill="none" stroke="#ffffff" strokeLinejoin="round" strokeWidth="5px" /></svg>
        <p className="back-to-search">Back to your search results</p>
      </div>

      <form className="filter-form">
        <div className="input-group">
          <label htmlFor="location" className="filter-label">
            Location
          </label>
          <input
            type="text"
            required
            id="location"
            name="location"
            className="filter-input"
            placeholder="Enter your location"
            onChange={handleChange}
            value={defaultValue.location}
          />
        </div>

        <div className="input-group">
          <label htmlFor="start" className="filter-label">
            Subscription starts within the
          </label>
          <select
            type="text"
            id="start"
            name="subscription_start_days"
            className="filter-input dropdown"
            onChange={handleChange}
            defaultValue={defaultValue.subscription_start_days}
          >
            <option value="2">Next 2 Days</option>
            <option value="14">Next 14 Days</option>
            <option value="30">Next 30 Days</option>
          </select>
        </div>

        <div className="input-group">
          <label htmlFor="distance" className="filter-label">
            Distance (radius in miles)
          </label>
          <select
            type="text"
            id="distance"
            name="max_distance"
            className="filter-input dropdown"
            onChange={handleChange}
            defaultValue={defaultValue.max_distance}
          >
            <option value="25">25</option>
            <option value="50">50</option>
            <option value="75">75</option>
            <option value="100">100</option>
            <option value="150">150</option>
            <option value="200">200</option>
            <option value="9999">Nationwide</option>
          </select>
        </div>

        <div className="input-group">
          <label htmlFor="budget" className="filter-label">
            Monthly Budget
          </label>
          <select
            type="text"
            id="budget"
            name="price_max"
            className="filter-input dropdown"
            onChange={handleChange}
            defaultValue={defaultValue.price_max}
          >
            <option value="200">Under £200</option>
            <option value="500">Under £500</option>
            <option value="750">Under £750</option>
            <option value="1000">Under £1000</option>
            <option value="1500">Under £1500</option>
            <option value="2000">Under £2000</option>
          </select>
        </div>

        <div className="input-group">
          <label htmlFor="gearbox" className="filter-label">
            Gearbox
          </label>
          <select
            type="text"
            id="gearbox"
            name="transmission"
            className="filter-input dropdown"
            onChange={handleChange}
            defaultValue={defaultValue.transmission}
          >
            <option value="any">Any</option>
            <option value="automatic">Automatic</option>
            <option value="cvt">Cvt</option>
            <option value="manual">Manual</option>
          </select>
        </div>

        <div className="input-group">
          <label htmlFor="seats" className="filter-label">
            Minimum number of seats
          </label>
          <select
            type="text"
            id="seats"
            name="number_of_seats_min"
            className="filter-input dropdown"
            onChange={handleChange}
            defaultValue={defaultValue.number_of_seats_min}
          >
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
            <option value="6">6</option>
            <option value="7">7</option>
            <option value="8">8</option>
            <option value="9">9</option>
          </select>
        </div>

        <div className="input-group">
          <label htmlFor="year" className="filter-label">
            Year
          </label>
          <select
            type="text"
            id="year"
            name="year"
            className="filter-input dropdown"
            onChange={handleChange}
            defaultValue={defaultValue.year}
          >
            <option value="any">Any</option>
            <option value="2018">2018</option>
            <option value="2017">2017</option>
            <option value="2016">2016</option>
            <option value="2015">2015</option>
            <option value="2014">2014</option>
            <option value="2013">2013</option>
            <option value="2012">2012</option>
          </select>
        </div>

        <div className="input-group">
          <label htmlFor="fuel" className="filter-label">
            Fuel Type
          </label>
          <select
            type="text"
            id="fuel"
            name="fuel"
            className="filter-input dropdown"
            onChange={handleChange}
            defaultValue={defaultValue.fuel}
          >
            <option value="any">Any</option>
            <option value="petrol">Petrol</option>
            <option value="diesel">Diesel</option>
            <option value="hybrid">Petrol Hybrid</option>
          </select>
        </div>
      </form>
    </div>
  );
}

Filter.propTypes = {
  toggle: func.isRequired,
  handleChange: func.isRequired,
  defaultValue: objectOf(any).isRequired,
  className: string.isRequired,
};
