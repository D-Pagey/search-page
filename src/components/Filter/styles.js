import styled from 'styled-components';

export const FilterForm = styled.form`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 0 auto 60px;
  padding: 0 1rem;
  width: 100%;
  @media (min-width: 768px) {
    width: 720px;
  }
  @media (min-width: 1024px) {
    width: initial;
}
`;

export const InputGroup = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 15px;
  width: 100%;
`;

export const FilterLabel = styled.label`
  color: #292b2c;
  font-size: 16px;
  font-weight: bold;
  margin: 0 0 .2rem 2px;
`;

export const Dropdown = styled.select`
  background-color: white;
  border: 1px solid #e4e3e3;
  border-radius: 6px;
  color: black;
  font-size: 100%;
  height: 40px;
  padding: 0 10px;
  width: 100%;
`;

export const Input = styled.input`
  background-color: white;
  border: 1px solid #e4e3e3;
  border-radius: 6px;
  color: black;
  font-size: 100%;
  height: 40px;
  padding: 0 10px;
  width: 100%;
`;

export const BackToSearchWrapper = styled.div`
  align-items: center;
  background-color: #172B24;
  display: flex;
  height: 52px;
  @media (min-width: 1024px) {
    display: none;
  }
`;

export const BackToSearchText = styled.p`
  color: white;
  margin-left: 10px;
`;

export const Chevron = styled.svg`
  margin-left: 20px;
`;

/* eslint-disable */
export const FilterWrapper = styled.div`
  display: ${props => props.hide ? 'none' : 'block'};
  width: 100%;
  @media (min-width: 1024px) {
    display: block;
    margin-left: 32px;
    width: 25%;
  }
`;
/* eslint-enable */
