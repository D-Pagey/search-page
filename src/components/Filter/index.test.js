import React from 'react';
import { shallow } from 'enzyme';
import Filter from '.';

it('should render', () => {
  const wrapper = shallow(<Filter />);
  expect(wrapper).toMatchSnapshot();
});
