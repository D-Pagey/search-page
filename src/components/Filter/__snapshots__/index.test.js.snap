// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="filter-container"
>
  <div
    className="back-to-search-div"
  >
    <svg
      className="chevron"
      data-name="Layer 4"
      height="20px"
      id="Layer_4"
      viewBox="0 0 23 43.41"
      width="20px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        white_chevron
      </title>
      <rect
        fill="none"
        height="83.33"
        rx="41.67"
        ry="41.67"
        width="83.33"
        x="-29.83"
        y="-19.96"
      />
      <polyline
        fill="none"
        points="21.17 1.71 11.83 11.71 2.5 21.71 11.83 31.71 21.17 41.71"
        stroke="#ffffff"
        strokeLinejoin="round"
        strokeWidth="5px"
      />
    </svg>
    <p
      className="back-to-search"
    >
      Back to your search results
    </p>
  </div>
  <form
    className="filter-form"
  >
    <div
      className="input-group"
    >
      <label
        className="filter-label"
        htmlFor="location"
      >
        Location
      </label>
      <input
        className="filter-input"
        id="location"
        name="location"
        placeholder="Enter your location"
        required={true}
        type="text"
      />
    </div>
    <div
      className="input-group"
    >
      <label
        className="filter-label"
        htmlFor="sub-start"
      >
        Subscription starts within the
      </label>
      <select
        className="filter-input dropdown"
        id="sub-start"
        name="sub-start"
        type="text"
      >
        <option
          value="2"
        >
          Next 2 Days
        </option>
        <option
          defaultValue={true}
          value="14"
        >
          Next 14 Days
        </option>
        <option
          value="30"
        >
          Next 30 Days
        </option>
      </select>
    </div>
    <div
      className="input-group"
    >
      <label
        className="filter-label"
        htmlFor="distance"
      >
        Distance (radius in miles)
      </label>
      <select
        className="filter-input dropdown"
        id="distance"
        name="distance"
        type="text"
      >
        <option
          defaultValue={true}
          value="25"
        >
          25
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value="75"
        >
          75
        </option>
        <option
          value="100"
        >
          100
        </option>
        <option
          value="150"
        >
          150
        </option>
        <option
          value="200"
        >
          200
        </option>
        <option
          value="all"
        >
          Nationwide
        </option>
      </select>
    </div>
    <div
      className="input-group"
    >
      <label
        className="filter-label"
        htmlFor="budget"
      >
        Monthly Budget
      </label>
      <select
        className="filter-input dropdown"
        id="budget"
        name="budget"
        type="text"
      >
        <option
          defaultValue={true}
          value="400"
        >
          Under £200
        </option>
        <option
          value="500"
        >
          Under £500
        </option>
        <option
          value="750"
        >
          Under £750
        </option>
        <option
          value="1000"
        >
          Under £1000
        </option>
        <option
          value="1500"
        >
          Under £1500
        </option>
        <option
          value="2000"
        >
          Under £2000
        </option>
      </select>
    </div>
    <div
      className="input-group"
    >
      <label
        className="filter-label"
        htmlFor="gearbox"
      >
        Gearbox
      </label>
      <select
        className="filter-input dropdown"
        id="gearbox"
        name="gearbox"
        type="text"
      >
        <option
          defaultValue={true}
          value="any"
        >
          Any
        </option>
        <option
          value="automatic"
        >
          Automatic
        </option>
        <option
          value="cvt"
        >
          Cvt
        </option>
        <option
          value="manual"
        >
          Manual
        </option>
      </select>
    </div>
    <div
      className="input-group"
    >
      <label
        className="filter-label"
        htmlFor="year"
      >
        Year
      </label>
      <select
        className="filter-input dropdown"
        id="year"
        name="year"
        type="text"
      >
        <option
          defaultValue={true}
          value="any"
        >
          Any
        </option>
        <option
          value="2018"
        >
          2018
        </option>
        <option
          value="2017"
        >
          2017
        </option>
        <option
          value="2016"
        >
          2016
        </option>
        <option
          value="2015"
        >
          2015
        </option>
        <option
          value="2014"
        >
          2014
        </option>
        <option
          value="2013"
        >
          2013
        </option>
        <option
          value="2012"
        >
          2012
        </option>
      </select>
    </div>
    <div
      className="input-group"
    >
      <label
        className="filter-label"
        htmlFor="fuel"
      >
        Fuel Type
      </label>
      <select
        className="filter-input dropdown"
        id="fuel"
        name="fuel"
        type="text"
      >
        <option
          defaultValue={true}
          value="any"
        >
          Any
        </option>
        <option
          value="petrol"
        >
          Petrol
        </option>
        <option
          value="diesel"
        >
          Diesel
        </option>
        <option
          value="hybrid"
        >
          Petrol Hybrid
        </option>
      </select>
    </div>
  </form>
</div>
`;
